### ISSUE TITLE
Broken tool tab styling (too many extensions?)

### ISSUE THREAD

**Issue by tsc-awardle** [mdsecresearch/BurpSuiteSharpener#72]
Hi there.  I've been using Sharpener for the past few months and absolutely love it.  However, I recently started seeing some weird behavior with the icon styling.

### Problem
Tool tab styling gets out of sync like in the following image.

![image](https://user-images.githubusercontent.com/67977601/169152154-62490760-b24c-484f-90cd-c62dccec6b2f.png)

In extreme cases, the tab text will disappear as well and can only be fixed by restarting Burp.

### Details
The issue is most prevalent when:
  1. Multiple extensions are loaded - specifically those that add their own JMenu to Burp's JMenuBar
  2. Icon themes are enabled
  3. Loading/unloading extensions after sharpener theme is applied

Tested on Burp Suite Professional v2022.3.7 (Windows 10 and Kali-Linux)

### Steps to reproduce
1. Load a few extensions into burp.  For example:
![image](https://user-images.githubusercontent.com/67977601/169155615-e67ec359-4cd3-46ea-849c-353c8c668593.png)
2. Enable a sharpener icon theme (for example: gradient)
3. If styling is not broken yet, try unloading/reloading some of the extensions or switch the icon theme on/off.  The icon themes usually break after doing this 2-3 times.

### Thoughts
This is just a hunch, but I suspect this is an issue with AWT/Swing concurrency.  Ideally, all GUI changes should be happening on a single event dispatch thread, but it seems like a race condition is causing two different updates to clash.

**Comment by irsdl** [comment]
Hi, that's great. Please feel free to submit a PR if it works with no problems after your tests. Please also test to see what would happen when you detach and attach a tool to ensure it can still put the icons back. 

The reason for a new thread is to not slow down Burp Suite app normal working process at any costs. That said, maybe we can run that thread with a delay perhaps?

**Comment by tsc-awardle** [comment]
Understood, I agree burp performance and responsiveness should be a top priority.

I'm still very new to GUI development in swing/AWT, but I found these articles very helpful. https://docs.oracle.com/javase/tutorial/uiswing/concurrency/index.html

The `invokeAndWait()` method already schedules the task asynchronously on the Event Dispatch Thread (EDT), so it won't block the current thread or other worker threads.  It should be perfectly equipped to handle all the GUI events in this extension.  It only becomes a problem if long-running tasks are scheduled on the EDT (ex: you wouldn't want to run a scanner thread here.)

The possibility of running it with a delay is interesting, but may come with it's own set of challenges. For example, I noticed that this scheduled task effectively kicks the thread outside the EDT.
![image](https://user-images.githubusercontent.com/67977601/169409378-7d92cb28-f914-4e94-8d7a-a0e7cf258a7f.png)
![image](https://user-images.githubusercontent.com/67977601/169409527-5e270873-16f5-420f-8d6c-44d33aaff03e.png)

This can be solved by wrapping the statement in another call to `invokeLater()`
![image](https://user-images.githubusercontent.com/67977601/169410027-91039595-ed9d-4fc2-978a-2b42b1afea41.png)

I made similar changes to other scheduled GUI operations.  Particularly, the `MainToolsTabWatcher.componentAdded()` as you mentioned that was a concern.  It worked fine before, but this should help ensure everything stays in sync.  It might even be possible to remove many of the hard-coded delays.

I've tested just about every feature and if anything, it's feels more responsive to me. Working on the PR now.
